app/controllers/api/imports.php

```diff
@@ -0,0 +1,403 @@
+    ->param('search', '', new Text(256), 'Search term to filter your list results. Max length: 256 chars.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (array $queries, string $search, Response $response, Database $dbForProject) {
+
+        $queries = Query::parseQueries($queries);
+
+        if (!empty($search)) {
+            $queries[] = Query::search('search', $search);
+        }
+
+        // Get cursor document if there was a cursor query
+        $cursor = Query::getByType($queries, Query::TYPE_CURSORAFTER, Query::TYPE_CURSORBEFORE);
+        $cursor = reset($cursor);
+        if ($cursor) {
+            /** @var Query $cursor */
+            $importId = $cursor->getValue();
+            $cursorDocument = $dbForProject->getDocument('imports', $importId);
+
+            if ($cursorDocument->isEmpty()) {
+                throw new Exception(Exception::GENERAL_CURSOR_NOT_FOUND, "Import '{$importId}' for the 'cursor' value not found.");
+            }
+
+            $cursor->setValue($cursorDocument);
+        }
+
+        $filterQueries = Query::groupByType($queries)['filters'];
+
+        $response->dynamic(new Document([
+            'imports' => $dbForProject->find('imports', $queries),
+            'total' => $dbForProject->count('imports', $filterQueries, APP_LIMIT_COUNT),
+        ]), Response::MODEL_IMPORT_LIST);
+    });
+
+App::get('/v1/imports/:importId')
+    ->groups(['api', 'imports'])
+    ->desc('Get Import')
+    ->label('scope', 'imports.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'get')
+    ->label('sdk.description', '/docs/references/imports/get-import.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_IMPORT)
+    ->param('importId', '', new UID(), 'Import unique ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $importId, Response $response, Database $dbForProject) {
+        $import = $dbForProject->getDocument('imports', $importId);
+
+        if ($import->isEmpty()) {
+            throw new Exception(Exception::IMPORT_NOT_FOUND, 'Import not found', 404);
+        }
+
+        $response->dynamic($import, Response::MODEL_IMPORT);
+    });
+
+App::post('/v1/imports/:importId')
+    ->groups(['api', 'imports'])
+    ->desc('Retry Import')
+    ->label('scope', 'imports.write')
+    ->label('event', 'imports.[importId].retry')
+    ->label('audits.event', 'import.retry')
+    ->label('audits.resource', 'imports/{request.importId}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'imports')
+    ->label('sdk.method', 'retry')
+    ->label('sdk.description', '/docs/references/imports/retry-import.md')
```

Comment:
1. Not sure this one should be POST vs PATCH
2. Let's ensure we use descriptions that are consistent with other services
3. Have we added md files for the docs?